packages/create-llama/templates/components/vectordbs/python/pg/utils.py

```diff
@@ -0,0 +1,68 @@
+import os
+from llama_index.vector_stores import PGVectorStore
+from pydantic import BaseModel
+from urllib.parse import urlparse
+
+
+class PGConnectionElements(BaseModel):
+    user: str
+    password: str
+    host: str
+    port: int
+    database: str
+
+
+def destruct_pg_connection_string(conn_string: str) -> PGConnectionElements:
```

Comment:
maybe add as static class method to `PGConnectionElements`? like `from_connection_string`?

Reply:
if the connection string can not be parsed, you can throw an exception (see comment below)